### PR TITLE
pdns-recursor: bump to 4.3.0 and fix build

### PR DESCRIPTION
**Pull request by alarcher**
No idea how to test this...

### CODE REVIEW

**Comment by alarcher** [comment]
@jeffpc Would you have any suggestion how to test this update?

**Comment by jeffpc** [comment]
I just "test it in production".

**Comment by iigs** [comment]
Making assumption from this page: https://blog.powerdns.com/2013/05/07/contributing-powerdns-tests-unit-tests/ , it looks like "gmake testrunner" should test the binary.  I don't use pdns-recursor.

This is the patch:

--- Makefile.orig       2020-05-27 20:58:01.934811556 +0000
+++ Makefile    2020-05-28 00:30:35.590121426 +0000
@@ -28,7 +28,6 @@
 COMPONENT_SUMMARY=     a DNS server
 COMPONENT_CLASSIFICATION=System/Services
 
-TEST_TARGET=$(NO_TESTS)
 include $(WS_MAKE_RULES)/common.mk
 
 CXXFLAGS+=-g -pthread
@@ -38,6 +37,8 @@
 COMPONENT_PRE_CONFIGURE_ACTION = \
        ($(CLONEY) $(SOURCE_DIR) $(@D))
 
+COMPONENT_TEST_TARGETS=testrunner
+
 CONFIGURE_SCRIPT = $(@D)/configure
 
 CONFIGURE_OPTIONS += --sysconfdir=/etc/powerdns


gmake test results in these errors:

In file included from /usr/include/boost/test/unit_test.hpp:19:0,
                 from test-dns_random_hh.cc:10:
test-dns_random_hh.cc:126:22: error: redefinition of 'struct test_dns_random_hh::test_dns_random_getrandom_average'
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
test-dns_random_hh.cc:106:22: note: previous definition of 'struct test_dns_random_hh::test_dns_random_getrandom_average'
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
In file included from /usr/include/boost/config.hpp:61:0,
                 from /usr/include/boost/test/test_tools.hpp:17,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from test-dns_random_hh.cc:10:
test-dns_random_hh.cc: In function 'void test_dns_random_hh::test_dns_random_getrandom_average_invoker()':
test-dns_random_hh.cc:126:22: error: redefinition of 'void test_dns_random_hh::test_dns_random_getrandom_average_invoker()'
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
test-dns_random_hh.cc:106:22: note: 'void test_dns_random_hh::test_dns_random_getrandom_average_invoker()' previously defined here
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
test-dns_random_hh.cc: At global scope:
test-dns_random_hh.cc:126:22: error: redefinition of 'struct test_dns_random_hh::test_dns_random_getrandom_average_id'
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
test-dns_random_hh.cc:106:22: note: previous definition of 'struct test_dns_random_hh::test_dns_random_getrandom_average_id'
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
In file included from /usr/include/boost/test/unit_test.hpp:19:0,
                 from test-dns_random_hh.cc:10:
test-dns_random_hh.cc:126:22: error: redefinition of 'void test_dns_random_hh::test_dns_random_getrandom_average::test_method()'
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
                      ^
test-dns_random_hh.cc:106:22: note: 'void test_dns_random_hh::test_dns_random_getrandom_average::test_method()' previously defined here
 BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {

File: components/network/pdns-recursor/patches/00-wontbuild.patch [hunk 1]
```diff
@@ -0,0 +1,101 @@
+--- pdns-recursor-4.3.0/ext/yahttp/yahttp/reqresp.cpp.orig	2020-05-17 13:22:48.488029417 +0000
```
Comment (pyhalov) [inline components/network/pdns-recursor/patches/00-wontbuild.patch:3]: Can you insert link to https://github.com/PowerDNS/pdns/pull/9031 ?

**Review by pyhalov — APPROVED** [review]

File: components/network/pdns-recursor/patches/00-wontbuild.patch (hunk 1 above)
Comment (alarcher) [reply components/network/pdns-recursor/patches/00-wontbuild.patch:3]: Done and I added the testsuite, all tests passed.

**Comment by alarcher** [comment]
> Making assumption from this page: https://blog.powerdns.com/2013/05/07/contributing-powerdns-tests-unit-tests/ , it looks like "gmake testrunner" should test the binary. I don't use pdns-recursor.
> 
> This is the patch:
> 
> --- Makefile.orig 2020-05-27 20:58:01.934811556 +0000
> +++ Makefile 2020-05-28 00:30:35.590121426 +0000
> @@ -28,7 +28,6 @@
> COMPONENT_SUMMARY= a DNS server
> COMPONENT_CLASSIFICATION=System/Services
> 
> -TEST_TARGET=$(NO_TESTS)
> include $(WS_MAKE_RULES)/common.mk
> 
> CXXFLAGS+=-g -pthread
> @@ -38,6 +37,8 @@
> COMPONENT_PRE_CONFIGURE_ACTION =
> ($(CLONEY) $(SOURCE_DIR) $(@d))
> 
> +COMPONENT_TEST_TARGETS=testrunner
> +
> CONFIGURE_SCRIPT = $(@d)/configure
> 
> CONFIGURE_OPTIONS += --sysconfdir=/etc/powerdns
> 
> gmake test results in these errors:
> 
> In file included from /usr/include/boost/test/unit_test.hpp:19:0,
> from test-dns_random_hh.cc:10:
> test-dns_random_hh.cc:126:22: error: redefinition of 'struct test_dns_random_hh::test_dns_random_getrandom_average'
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> test-dns_random_hh.cc:106:22: note: previous definition of 'struct test_dns_random_hh::test_dns_random_getrandom_average'
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> In file included from /usr/include/boost/config.hpp:61:0,
> from /usr/include/boost/test/test_tools.hpp:17,
> from /usr/include/boost/test/unit_test.hpp:18,
> from test-dns_random_hh.cc:10:
> test-dns_random_hh.cc: In function 'void test_dns_random_hh::test_dns_random_getrandom_average_invoker()':
> test-dns_random_hh.cc:126:22: error: redefinition of 'void test_dns_random_hh::test_dns_random_getrandom_average_invoker()'
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> test-dns_random_hh.cc:106:22: note: 'void test_dns_random_hh::test_dns_random_getrandom_average_invoker()' previously defined here
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> test-dns_random_hh.cc: At global scope:
> test-dns_random_hh.cc:126:22: error: redefinition of 'struct test_dns_random_hh::test_dns_random_getrandom_average_id'
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> test-dns_random_hh.cc:106:22: note: previous definition of 'struct test_dns_random_hh::test_dns_random_getrandom_average_id'
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> In file included from /usr/include/boost/test/unit_test.hpp:19:0,
> from test-dns_random_hh.cc:10:
> test-dns_random_hh.cc:126:22: error: redefinition of 'void test_dns_random_hh::test_dns_random_getrandom_average::test_method()'
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {
> ^
> test-dns_random_hh.cc:106:22: note: 'void test_dns_random_hh::test_dns_random_getrandom_average::test_method()' previously defined here
> BOOST_AUTO_TEST_CASE(test_dns_random_getrandom_average) {

Thank you :) I fixed the testsuite based on your input :)

**Pull request merged**